src/generate-ova-from-template.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+
+import struct
+import tempfile
+import os
+import sys
+import json
+from subprocess import check_call
+from stat import *
+from shutil import rmtree, move
+import tarfile
+
+# Usage: generate-ova-from-template.py input_image.qcow2 input_template.xml example_ova_parameters.json output_image.ova
+
+def qcow_disk_size(filename):
+    # Detect if an image is in qcow format
+    # If it is, return the size
+    # If it isn't, return None
+    # Borrowed from Image Factory and modified
+
+    # For interested parties, this is the QCOW header struct in C
+    # struct qcow_header {
+    #    uint32_t magic;
+    #    uint32_t version;
+    #    uint64_t backing_file_offset;
+    #    uint32_t backing_file_size;
+    #    uint32_t cluster_bits;
+    #    uint64_t size; /* in bytes */
+    #    uint32_t crypt_method;
+    #    uint32_t l1_size;
+    #    uint64_t l1_table_offset;
+    #    uint64_t refcount_table_offset;
+    #    uint32_t refcount_table_clusters;
+    #    uint32_t nb_snapshots;
+    #    uint64_t snapshots_offset;
+    # };
+
+    # And in Python struct format string-ese
+    qcow_struct=">IIQIIQIIQQIIQ" # > means big-endian
+    qcow_magic = 0x514649FB # 'Q' 'F' 'I' 0xFB
+
+    f = open(filename,"rb")
+    pack = f.read(struct.calcsize(qcow_struct))
+    f.close()
+
+    unpack = struct.unpack(qcow_struct, pack)
+
+    if unpack[0] == qcow_magic:
+        return unpack[5]
```

Comment:
nit: I believe this maps to `uint64_t size` but a note here would be great for future reading.